### ISSUE TITLE
Trouble reading / writing CVs using JMRI

### ISSUE THREAD

**Issue by mgagneur** [habazut/dcc-ardu#1]
Hello Harold,
I have been following the discussion on Trainboard and am very happy the team of talented coders is updating DCC++.

I downloaded your code on March 28 and using JMRI (4.18) everything works perfectly. I can read and write cvs and change the loco long address with no issues. Works perfectly.

Today I downloaded an update and cannot read or write. I receive the dreaded 308 error.
If I then reload the code from March 28 everything works again. It is 100 % repeatable.

Thank you for the effort you are putting into this.

Best regards,

Marty

**Comment by habazut** [comment]
That's odd because the changes between March 28 "1.3.0+haba" and today "11.0.0+haba" are not in the programming code for the programming track. In spite of that I did make a big version number jump to avoid confusion with DCC++ classic versions the changes are quite small. 
See: git diff d141fb4 478cef7

But let's try to find the problem...

What hardware are you on? I have here the Uno and the MotorShield Rev3. I can read CV both from JMRI and directly by connecting to the serial line like:

```
[connect to /dev/ttyUSB1 with 115000 baud]
<iDCC++ BASE STATION FOR ARDUINO<iDCC++ BASE STATION FOR ARDUINO UNO / ARDUINO MOTOR SHIELD: V-11.0.0+haba / Apr  4 2020 11:11:05><n0: SERIAL >
 <L>  [I type, not echoed]
currentReg: 762 recycleReg: 0 maxLoadedReg: 762
Slot:   Reg     Bits
M0:     754     0       F_0
M1:     762     28      7F      80      1F      F0      F_0

currentReg: 1879 recycleReg: 0 maxLoadedReg: 1879
Slot:   Reg     Bits
P0:     1871    0       F_3
P1:     1879    28      7F      80      1F      F0      F_0
 <R 1 1 1>  [I type, not echoed]
[prog track power on, loco motor acks, prog power off]
<r1|1|1 3>
```

So CV1 = 3. Good.

How does that look at your setup?

The thing I changed between March 28 and today is that the code enables pin 2 of the Arduino to give a trigger signal. You can disable this change by editing Config.h and changing

#define USE_TRIGGERPIN

to 

//#define USE_TRIGGERPIN

Let me know if that makes any difference.

Then apart from reading and writing on the prog track, can you run trains on the main track with the code from today?

Regards,
Harald.

PS: I don't like markdown ;)

**Comment by mgagneur** [comment]
Hello Harald,

Thank you for the note.
I am using an Arduino MEGA and Arduino MotorShield Rev3.

I tried several locos (2 with Bachmann Sound Value and 1 with ESU
Loksound). With the earlier rev all programmed with no issue. With the
newer rev all failed on the programming track but ran on the OPS track.

With the newer rev I received:
<iDCC++ BASE STATION FOR ARDUINO MEGA / ARDUINO MOTOR SHIELD: V-11.0.0+haba
/ Apr  6 2020 13:02:37><n0: SERIAL

typing in <R 1 1 1> returned <r1|1|1 3> for the earlier rev and <r1|1|1 -1>
for the newer

When I commented out //#define USE_TRIGGERPIN

*The newer version worked perfectly:)*

I think I see now. When I look at the pin mappings drawing pin 2 is free on
the Uno but is used on the Mega. I probably simply missed where you
mentioned to watch out for that.

Regards,

Marty
